src/hotspot/share/utilities/elfFile.hpp

```diff
@@ -201,17 +204,569 @@ class ElfFile: public CHeapObj<mtInternal> {
+ *     The DWARF file is then opened by calling the constructor of this class. Once this is done, the processing of the
+ *     DWARF file is initiated by calling find_filename_and_line_number().
+ * (2) Find the compilation unit offset by reading entries from the section .debug_aranges, which contain address range
+ *     descriptors, until we find the correct descriptor that includes the library offset.
+ * (3) Find the .debug_line offset for the line number information program from the .debug_info section:
+ *     (a) Parse the compilation unit header from the .debug_info section at the offset obtained by (2).
+ *     (b) Read the debug_abbrev_offset into the .debug_abbrev section that belongs to this compilation unit from the
+ *         header obtained in (3a).
+ *     (c) Read the abbreviation code that immediately follows the compilation unit header from (3a) which is needed to
+ *         find the correct entry in the .debug_abbrev section.
+ *     (d) Find the correct entry in the abbreviation table in the .debug_abbrev section by starting to parse entries at
+ *         the debug_abbrev_offset from (3b) until we find the correct one matching the abbreviation code from (3c).
+ *     (e) Read the specified attributes of the abbreviation entry from (3d) from the compilation unit (in the .debug_info
+ *         section) until we find the attribute DW_AT_stmt_list. This attributes represents an offset into the .debug_line
+ *         section which contains the line number program information to get the filename and the line number.
+ *  (4) Find the filename and line number belonging to the given library offset by running the line number program state
+ *      machine with its registers. This creates a matrix where each row stores information for specific addresses (library
+ *      offsets). The state machine executes different opcodes which modify the state machine registers. Certain opcodes
+ *      will add a new row to the matrix by taking the current values of state machine registers. As soon as the correct
+ *      matrix row matching the library offset is found, we can read the line number from the line register of the state
+ *      machine and parse the filename from the line number program header with the given file index from the file register
+ *      of the state machine.
+ *
+ *  More details about the different phases can be found at the associated classes and methods. A visualization of the
+ *  algorithm inside the different sections can be found in the class comments for DebugAranges, DebugAbbrev and
+ *  LineNumberProgram further down in this file.
+ *
+ *  Available log levels (-Xlog:dwarf=X):
+ *  - info:  Prints the path of parsed DWARF file together with the query and the resulting source information.
+ *  - debug: Prints the results of the steps (1) - (4) together with the generated line information matrix.
+ *  - trace: Full logging information for intermediate states/results when parsing the DWARF file.
+ */
```

Comment:
Maybe add a comment that log output is only supported in non-product builds and the reason.